repl/src/main/scala/com/cloudera/livy/repl/Session.scala

```diff
@@ -106,20 +100,34 @@ class Session(interpreter: Interpreter)
     _statements.clear()
   }
 
+  private def changeState(newState: SessionState): Unit = {
+    synchronized {
```

Comment:
Do we need to check whether previous state is the same as new state, so we don't need to update.